server/message_hooks.go

```diff
@@ -515,9 +524,11 @@ func (p *Plugin) SendChat(srcUser string, usersIDs []string, post *model.Post) (
 		attachment, err = client.UploadFile("", "", fileName+"_"+fileInfo.Id+fileExtension, int(fileInfo.Size), fileInfo.MimeType, bytes.NewReader(fileData), chat)
 		if err != nil {
 			p.API.LogWarn("Error in uploading file attachment to MS Teams", "error", err)
+			p.metricsService.ObserveFilesCount(handlers.ActionCreated, actionSourceMattermost, handlers.DirectMessageTrue, discardedReasonUnableToUploadFileOnTeams, int64(handlers.IncreaseFileCountByOne))
```

Comment:
If we are not able to successfully upload the file on Teams, should we still increment the files count by one?

Reply:
yes, becuase we are also adding a discarded reason which will denote the count of files discared due to a particular reason